core/pom.xml

```diff
@@ -166,8 +166,8 @@
       <plugin>
         <artifactId>maven-compiler-plugin</artifactId>
         <configuration>
-          <source>1.7</source>
-          <target>1.7</target>
+          <source>1.8</source>
```

Comment:
why change this in this pr? all modules are using 1.7 as default version, it's better raise another pr to handle this ?

Reply:
OK, I know why change this here.
Will Carbondata use 1.8 as default version?

Reply:
I think it is better to change it as java 1.8 provides some good features that makes code cleaner